### PR TITLE
Waterworks Port & Water Tweaks

### PR DESCRIPTION
**Pull request by goober3**
## About The Pull Request
ports https://github.com/tgstation/tgstation/pull/52865 (barring plumbing-enabled hydroponics)

and associated tweaks & fixes 
https://github.com/tgstation/tgstation/pull/53279
https://github.com/tgstation/tgstation/pull/53377
https://github.com/tgstation/tgstation/pull/54236
https://github.com/tgstation/tgstation/pull/55895
https://github.com/tgstation/tgstation/pull/53311

water can now be scooped from water tiles again (this was commented out in #1900)
water scooped from unpotable sources must be treated via boiling to become safe to drink.

## Why It's Good For The Game

image
punished for his sins.

## Changelog

:cl: ArcaneMusic, TemporalOroboros, Jack7D1, ported by goober3
add: sinks & showers are now compatible with plumbing, and have an internal water supply.
add: metal sheets can be used on reagent tanks to convert them into a plumbing-enabled tank.
add: water recyclers are now used to construct sinks & showers. print them at an autolathe.
fix: water can be sourced from water tiles again.
tweak: water from unpotable sources must be treated via boiling to become safe to drink.
/:cl:

### CODE REVIEW

File: code/game/objects/items/storage/boxes.dm [hunk 1]
```diff
@@ -1556,3 +1556,15 @@
 		/obj/item/stack/wrapping_paper/small=1
 		)
 	generate_items_inside(items_inside,src)
+
+/obj/item/storage/box/plumbing
```
Comment (Erikafox) [inline code/game/objects/items/storage/boxes.dm:1560]: you should add this as a relatively cheap cargo pack

File: code/game/objects/structures/shower.dm [hunk 2]
```diff
@@ -1,38 +1,61 @@
...
 #define SHOWER_NORMAL "normal"
+#define SHOWER_NORMAL_TEMP 300
 #define SHOWER_BOILING "boiling"
+#define SHOWER_BOILING_TEMP 400
+#define SHOWER_REACTION_MULTIPLIER 0.025
 
 /obj/machinery/shower
 	name = "shower"
-	desc = "The HS-451. Installed in the 2550s by the Nanotrasen Hygiene Division."
+	desc = "The HS-452. Installed in the 2550s by the Nanotrasen Hygiene Division, now with 2560 lead compliance! Passively replenishes itself with water when not in use." //this'll probably get a more lore compliant description later
```
Comment (Erikafox) [inline code/game/objects/structures/shower.dm]: Needs setting appropriate flavor

File: code/game/turfs/open/water.dm [hunk 3]
```diff
@@ -8,30 +8,29 @@
...
 	clawfootstep = FOOTSTEP_WATER
 	heavyfootstep = FOOTSTEP_WATER
 
-	var/datum/reagent/reagent_to_extract = /datum/reagent/water
+	var/datum/reagent/reagent_to_extract = /datum/reagent/water/dirty
 	var/extracted_reagent_visible_name = "water"
 
-/*
+
 /turf/open/water/attackby(obj/item/tool, mob/user, params)
```
Comment (Erikafox) [inline code/game/turfs/open/water.dm:22]: I remember why it was commented out. You have two attackbys declared, one for fish, one for not fish. There should only be 1 attackby. Take a look at another object that has multiple interactions on attack for an example

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 4]
```diff
@@ -298,13 +298,31 @@
...
+	color = "#908982"
+	taste_description = "swamp"
+	glass_icon_state = "icedcoffeeglass" //mmm yumy iced coffee
+	glass_name = "glass of water"
+	glass_desc = "The father of all refre- Ew. It's full of bacteria."
+	shot_glass_icon_state = "shotglassbrown"
+
+/datum/reagent/water/dirty/on_mob_life(mob/living/carbon/M)
+	M.adjustToxLoss(0.5, 0)
+	M.hallucination += 2
```
Comment (Erikafox) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: I wouldn't do hallucinations honestly. The rest seems normal-ish.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 5]
```diff
@@ -298,13 +298,31 @@
...
+	M.jitteriness += 1
+	M.disgust += 1
+	..()
+	return TRUE
+
 /datum/reagent/hydrogen_peroxide
```
Comment (Erikafox) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:322]: you should add a recipe for hydrogen peroxide, give it a minor wound tending bonus, and also 
https://pubmed.ncbi.nlm.nih.gov/15298493/
ingestion effects.

File: code/game/objects/items/storage/boxes.dm (hunk 1 above)
Comment (goober3) [reply code/game/objects/items/storage/boxes.dm:1560]: exactly what i'm working on right now.

File: code/game/objects/structures/shower.dm (hunk 2 above)
Comment (goober3) [reply code/game/objects/structures/shower.dm]: oh i forgot about that. oops

**Comment by goober3** [comment]
i'll finish the rest of this tomorrow.

File: code/game/objects/structures/shower.dm [hunk 6]
```diff
@@ -1,38 +1,61 @@
...
 /obj/machinery/shower/Initialize()
 	. = ..()
-	create_reagents(reaction_volume)
-	reagents.add_reagent(reagent_id, reaction_volume)
-
+	create_reagents(reagent_capacity)
+	reagents.add_reagent(reagent_id, reagent_capacity)
 	soundloop = new(list(src), FALSE)
 
+	AddComponent(/datum/component/plumbing/simple_demand)
```
Comment (tmtmtl30) [inline code/game/objects/structures/shower.dm:39]: I'm not a big fan of the plumbing integration, as discussed. It doesn't fit very well on shiptest. The other changes are fine; if they're isolated I think the PR will be fine.

**Review by tmtmtl30 — CHANGES_REQUESTED** [review]

**Comment by goober3** [comment]
may perhaps potentially redo this in another form!! without plumbing. we'll see

**Pull request closed without merging**